### PR TITLE
Handle bad responses from Hydra

### LINKED ISSUE #16: HTTP 409 on accepting login challenge
There is some weird issue with accepting logins going on. The log is spammed with errors like:

```
2021-09-28 10:31:03.602 ERROR 1 --- [r-http-epoll-11] c.f.u.error.GlobalExceptionHandler       : Unknown internal error occurred.
org.springframework.web.reactive.function.client.WebClientResponseException$Conflict: 409 Conflict from PUT http://faf-ory-hydra:4445/oauth2/auth/requests/login/accept?login_challenge=xxx
        at org.springframework.web.reactive.function.client.WebClientResponseException.create(WebClientResponseException.java:208) ~[spring-webflux-5.3.9.jar:5.3.9]
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
        |_ checkpoint ⇢ 409 from PUT http://faf-ory-hydra:4445/oauth2/auth/requests/login/accept?login_challenge=xxx [DefaultWebClient]
Stack trace:
                at org.springframework.web.reactive.function.client.WebClientResponseException.create(WebClientResponseException.java:208) ~[spring-webflux-5.3.9.jar:5.3.9]
                at org.springframework.web.reactive.function.client.DefaultClientResponse.lambda$createException$1(DefaultClientResponse.java:213) ~[spring-webflux-5.3.9.jar:5.3.9]
                at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxDefaultIfEmpty$DefaultIfEmptySubscriber.onNext(FluxDefaultIfEmpty.java:101) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:127) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:295) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:337) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:159) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:142) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:260) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:142) ~[reactor-core-3.4.9.jar:3.4.9]
                at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:400) ~[reactor-netty-core-1.0.10.jar:1.0.10]
                at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:419) ~[reactor-netty-core-1.0.10.jar:1.0.10]
                at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:473) ~[reactor-netty-core-1.0.10.jar:1.0.10]
                at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:684) ~[reactor-netty-http-1.0.10.jar:1.0.10]
                at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:93) ~[reactor-netty-core-1.0.10.jar:1.0.10]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795) ~[netty-transport-native-epoll-4.1.67.Final-linux-x86_64.jar:4.1.67.Final]
                at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480) ~[netty-transport-native-epoll-4.1.67.Final-linux-x86_64.jar:4.1.67.Final]
                at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.67.Final-linux-x86_64.jar:4.1.67.Final]
                at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.67.Final.jar:4.1.67.Final]
                at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.67.Final.jar:4.1.67.Final]
                at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
```

### PR DESCRIPTION
**Pull request by Brutus5000**
Fixes #16

### CODE REVIEW

File: src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt [hunk 1]
```diff
@@ -40,28 +42,31 @@ class OAuthController(
...
-        headers.location = URI.create(uriString)
-    }.setComplete()
+    ): Mono<Rendering> =
+        hydraService.getLoginRequest(challenge)
+            .flatMap {
+                val loginFailed = request.queryParams.containsKey("login_failed")
+                val loginThrottled = request.queryParams.containsKey("login_throttled")
+                model.addAttribute("loginFailed", loginFailed)
+                model.addAttribute("loginThrottled", loginThrottled)
+                model.addAttribute("challenge", challenge)
```
Comment (Sheikah45) [inline src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt:61]: why not populate with the challenge returned by hydra here?

File: src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt [hunk 2]
```diff
@@ -104,13 +109,15 @@ class OAuthController(
...
                         )
+                        is LoginResult.TechnicalError -> redirectTechnicalError(response)
                     }
                 }
         }
 
     @GetMapping("/consent")
     fun showConsent(
         request: ServerHttpRequest,
+        response: ServerHttpResponse,
```
Comment (Sheikah45) [inline src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt:120]: Was this added just for completeness it doesn't seem used

Comment (Brutus5000) [reply src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt]: Yeah there was a lot of stuff to clean up.

File: src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt (hunk 1 above)
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt:61]: But we don't need anything from this object.

**Pull request merged**